rusk/src/lib/chain/rusk.rs

```diff
@@ -122,6 +137,7 @@ impl Rusk {
                     // re-execute all spent transactions. We don't discard the
                     // transaction, since it is technically valid.
                     if gas_spent > block_gas_left {
+                        warn!("This is not supposed to happen with conservative tx inclusion");
```

Comment:
why not panic? This clearly looks like a bug to me (unless there are some unpredictable gas fluctuations?)

Reply:
Because I don't want to stop the network if something is wrong.

A panic will let the block to never being produced